components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -233,7 +233,8 @@ private boolean validateCallbackURI(String callbackURI, OAuthAppDO oauthApp) {
         String regexp = null;
         String registeredCallbackUrl = oauthApp.getCallbackUrl();
         if (registeredCallbackUrl.startsWith(OAuthConstants.CALLBACK_URL_REGEXP_PREFIX)) {
-            regexp = registeredCallbackUrl.substring(OAuthConstants.CALLBACK_URL_REGEXP_PREFIX.length());
+            regexp = (registeredCallbackUrl.substring(OAuthConstants.CALLBACK_URL_REGEXP_PREFIX.length()))
+                    .replace("?", "\\?");
```

Comment:
Can we check if there are any inbuilt java methods to support this instead of a string replace.
Also what would happen if there is an actual regex containing `?` is here?
i.e. when registering the callback how do we differentiate whether the `?` is due to URL containing query params or due to being part of a regex pattern?

Reply:
Changed to replace only the first occurance of "?"